backend/app/core/storage.py

```
@@ -0,0 +1,99 @@
+        pass
+
+class MockStorage(StorageInterface):
+    """
+    Mock implementation of the StorageInterface.
+    Simulates storage operations without actual network calls.
+    Uses an in-memory dictionary to simulate storage for more robust mocking if needed,
+    but primarily logs actions and returns predefined values.
+    """
+    def __init__(self, bucket_name: str = "mock-bucket"):
+        self.bucket_name = bucket_name
+        self.mock_db: Dict[str, bytes] = {} # In-memory storage
```

Ruff (0.11.9)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)